src/config-import.js

```diff
@@ -0,0 +1,271 @@
+            preset.sort = 100
+          }
+          return preset
+        })
+        .sort((preset, nextPreset) => nextPreset.sort - preset.sort)
+
+      // 5. for each preset get the corresponding fieldId and iconId, add them to the db
+      for (let preset of sortedPresets) {
+        /** @type {any} */
+        const presetValue = {
+          schemaName: 'preset',
+          fieldIds: [],
+          addTags: {},
+          removeTags: {},
+          terms: [],
+        }
+        for (const key of Object.keys(valueSchemas.preset.properties)) {
+          if (key in preset) {
+            // @ts-ignore - we validate below
+            presetValue[key] = preset[key]
+          }
+        }
+        if (!validate('preset', presetValue)) {
+          warnings.push(new Error(`Invalid preset ${preset.name}`))
+          continue
+        }
+        yield {
+          fieldNames:
+            'fields' in preset && Array.isArray(preset.fields)
+              ? preset.fields
+              : [],
+          iconName:
+            'icon' in preset && typeof preset.icon === 'string'
+              ? preset.icon
+              : undefined,
+          value: presetValue,
+        }
+      }
+    },
+  }
+}
+
+/**
+ * @param {string} filename
+ * @param {Buffer} buf
+ * @returns {{ name: string, variant: IconData['variants'][Number] }}}
+ */
+function parseIcon(filename, buf) {
+  const parsedFilename = path.parse(filename)
+  const matches = parsedFilename.base.match(
+    /([a-zA-Z0-9-]+)-([a-zA-Z]+)@(\d+)x\.(png|svg)$/
+  )
+  if (!matches) {
+    throw new Error(`Unexpected icon filename ${filename}`)
+  }
+  /* eslint-disable no-unused-vars */
+  const [_, name, size, pixelDensityStr, extension] = matches
+  const pixelDensity = Number(pixelDensityStr)
+  if (!(pixelDensity === 1 || pixelDensity === 2 || pixelDensity === 3)) {
+    throw new Error(`Error loading icon. invalid pixel density ${pixelDensity}`)
+  }
+  if (!(size === 'small' || size === 'medium' || size === 'large')) {
+    throw new Error(`Error loading icon. invalid size ${size}`)
+  }
+  if (!name) {
+    throw new Error('Error loading icon. missing name')
+  }
+  /** @type {'image/png' | 'image/svg+xml'} */
+  let mimeType
+  switch (parsedFilename.ext.toLowerCase()) {
+    case '.png':
+      mimeType = 'image/png'
+      break
+    case '.svg':
+      mimeType = 'image/svg+xml'
+      break
+    default:
+      throw new Error(`Unexpected icon extension ${parsedFilename.ext}`)
+  }
+  return {
+    name,
+    variant: {
+      size,
+      mimeType,
+      pixelDensity,
+      blob: buf,
+    },
+  }
+}
+
+/**
+ * @param {unknown} presetsFile
+ * @returns {asserts presetsFile is PresetsFile}
+ */
+function validatePresetsFile(presetsFile) {
+  if (
+    !isRecord(presetsFile) ||
+    !('presets' in presetsFile) ||
+    typeof presetsFile.presets !== 'object' ||
+    !('fields' in presetsFile) ||
+    typeof presetsFile.fields !== 'object'
```

Comment:
Checking for `!== 'object'` could return false positives for `null` or arrays. Maybe we could use `isRecord` instead?

```suggestion
    !isRecord(presetsFile) ||
    !isRecord(presetsFile.presets) ||
    !isRecord(presetsFile.fields)
```